ext/storage_sources/s3_store/s3_log_system.h

```diff
@@ -13,22 +43,48 @@ static const std::map<int32_t, Aws::Utils::Logging::LogLevel> verbosityMapping =
   {-1, Aws::Utils::Logging::LogLevel::Info}, {0, Aws::Utils::Logging::LogLevel::Info},
   {1, Aws::Utils::Logging::LogLevel::Debug}};
 
+// This class provides the S3 Store with a logger implementation that redirects the generated
+// logs to WiredTiger's logging streams. This class implements AWS's LogSystemInterface class,
+// an interface for logging implementations. This class derives functions from the interface to
+// incorporate the logging with WiredTiger's logging system.
 class S3LogSystem : public Aws::Utils::Logging::LogSystemInterface {
-
     public:
     S3LogSystem(WT_EXTENSION_API *wtApi, uint32_t wtVerbosityLevel);
     Aws::Utils::Logging::LogLevel
     GetLogLevel(void) const override
     {
         return (_awsLogLevel);
     }
+
+    // This function is inherited from AWS's LogSystemInterface. This function does a printf style
+    // output to the output stream. This function is mostly unchanged from AWS' implementation.
     void Log(
       Aws::Utils::Logging::LogLevel logLevel, const char *tag, const char *format, ...) override;
+
+    // This function is inherited from AWS's LogSystemInterface. This function writes the log stream
+    // to the output stream, in this case WiredTiger's output stream.
     void LogStream(Aws::Utils::Logging::LogLevel logLevel, const char *tag,
       const Aws::OStringStream &messageStream) override;
-    void LogErrorMessage(const std::string &message) const;
-    void LogDebugMessage(const std::string &message) const;
+
+    // This function sends error messages to WiredTiger's error level log stream.
```

Comment:
We dont have to say in comments that "This function ...". We could assume we are talking about the next function, and say "Send error messages ..". Same for the following comments.